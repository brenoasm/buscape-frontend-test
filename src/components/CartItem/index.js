import React, { Fragment } from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

import { colors } from "../../theme";

import CartSubtotal from '../CartSubtotal';
import Image from '../Image';

const propTypes = {
  cartItem: PropTypes.object.isRequired,
  handleProductRemoveClick: PropTypes.func,
  onBrokenImageUrl: PropTypes.func.isRequired
};

const defaultProps = {
  cartItem: null,
  handleProductRemoveClick: () => {},
  onBrokenImageUrl: () => {}
};

const StyledCartItem = styled.div`
  margin-top: 2px;
  padding: 20px;
  background-color: ${colors.toolbarItems};
  align-self: stretch;
  display: flex;

  > div: last-child {
    display: flex;
    width: 100%;

    > div:first-child {
      display: flex;
      flex-direction: column;
      justify-content: space-between;
      padding: 0 10px;
      line-height: 0.9em;
      font-weight: 700;
      height: 100%;

      > span {
        color: ${colors.white};
      }
    }

    > div:last-child {
      margin-left: auto;
      padding: 0 10px;

      > i {
        height: 15px;
      }
    }
  }
`;

const CartItem = ({ cartItem, handleProductRemoveClick, onBrokenImageUrl }) => {
  return (
    <Fragment>
      {cartItem && (
        <StyledCartItem>
          <Image
            src={cartItem.imageUrl}
            alt="Produto"
            onBrokenImageUrl={onBrokenImageUrl}
            height={80}
            width={80}
          />
          <div>
            <div>
              <span>{cartItem.name}</span>
              <CartSubtotal price={cartItem.price} />
            </div>
            <div onClick={() => handleProductRemoveClick(cartItem)}>
              <i className="fas fa-times" />
            </div>
          </div>
        </StyledCartItem>
      )}
    </Fragment>
  );
};

CartItem.propTypes = propTypes;
CartItem.defaultProps = defaultProps;

export default CartItem;
