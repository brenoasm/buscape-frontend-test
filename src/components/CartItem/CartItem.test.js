import React from "react";
import { shallow } from "enzyme";

import CartItem from "./";

describe("CartItem", () => {
  const cartItem = {
    id: 2321312,
    imageUrl:
      "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
    name: "Smartphone Apple iPhone 7 128GB",
    price: { installmentValue: 389.9, installments: 10, value: 3509.1 }
  };

  it("matches the snapshot", () => {
    const wrapper = shallow(<CartItem cartItem={cartItem} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("calls the fn once when clicked", () => {
    const fn = jest.fn();
    const wrapper = shallow(<CartItem cartItem={cartItem} handleProductRemoveClick={fn} />);
    wrapper.find('div[data-test-button]').simulate("click");
    expect(fn).toBeCalled();
  });
});
