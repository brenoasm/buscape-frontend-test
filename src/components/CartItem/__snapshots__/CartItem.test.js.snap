// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CartItem
    cartItem={
      Object {
        "id": 2321312,
        "imageUrl": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
        "name": "Smartphone Apple iPhone 7 128GB",
        "price": Object {
          "installmentValue": 389.9,
          "installments": 10,
          "value": 3509.1,
        },
      }
    }
    handleProductRemoveClick={[Function]}
    onBrokenImageUrl={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <ForwardRef>
        <Image
          alt="Produto"
          height={80}
          onBrokenImageUrl={[Function]}
          src="http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg"
          width={80}
        />
        <div>
          <div>
            <span>
              Smartphone Apple iPhone 7 128GB
            </span>
            <CartSubtotal
              price={
                Object {
                  "installmentValue": 389.9,
                  "installments": 10,
                  "value": 3509.1,
                }
              }
            />
          </div>
          <div
            data-test-button="remove-button"
            onClick={[Function]}
          >
            <i
              className="fas fa-times"
            />
          </div>
        </div>
      </ForwardRef>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Image
            alt="Produto"
            height={80}
            onBrokenImageUrl={[Function]}
            src="http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg"
            width={80}
          />,
          <div>
            <div>
              <span>
                Smartphone Apple iPhone 7 128GB
              </span>
              <CartSubtotal
                price={
                  Object {
                    "installmentValue": 389.9,
                    "installments": 10,
                    "value": 3509.1,
                  }
                }
              />
            </div>
            <div
              data-test-button="remove-button"
              onClick={[Function]}
            >
              <i
                className="fas fa-times"
              />
            </div>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "alt": "Produto",
            "height": 80,
            "onBrokenImageUrl": [Function],
            "src": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
            "width": 80,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <span>
                  Smartphone Apple iPhone 7 128GB
                </span>
                <CartSubtotal
                  price={
                    Object {
                      "installmentValue": 389.9,
                      "installments": 10,
                      "value": 3509.1,
                    }
                  }
                />
              </div>,
              <div
                data-test-button="remove-button"
                onClick={[Function]}
              >
                <i
                  className="fas fa-times"
                />
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    Smartphone Apple iPhone 7 128GB
                  </span>,
                  <CartSubtotal
                    price={
                      Object {
                        "installmentValue": 389.9,
                        "installments": 10,
                        "value": 3509.1,
                      }
                    }
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Smartphone Apple iPhone 7 128GB",
                  },
                  "ref": null,
                  "rendered": "Smartphone Apple iPhone 7 128GB",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "price": Object {
                      "installmentValue": 389.9,
                      "installments": 10,
                      "value": 3509.1,
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="fas fa-times"
                />,
                "data-test-button": "remove-button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-times",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-htpNat",
          "isStatic": true,
          "rules": Array [
            "
  margin-top: 2px;
  padding: 20px;
  background-color: ",
            "#DAAF00",
            ";
  align-self: stretch;
  display: flex;

  > div: last-child {
    display: flex;
    width: 100%;

    > div:first-child {
      display: flex;
      flex-direction: column;
      justify-content: space-between;
      padding: 0 10px;
      line-height: 0.9em;
      font-weight: 700;
      height: 100%;

      > span {
        color: ",
            "#FFF",
            ";
      }
    }

    > div:last-child {
      margin-left: auto;
      padding: 0 10px;
      cursor: pointer;
    }
  }
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-htpNat",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ForwardRef>
          <Image
            alt="Produto"
            height={80}
            onBrokenImageUrl={[Function]}
            src="http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg"
            width={80}
          />
          <div>
            <div>
              <span>
                Smartphone Apple iPhone 7 128GB
              </span>
              <CartSubtotal
                price={
                  Object {
                    "installmentValue": 389.9,
                    "installments": 10,
                    "value": 3509.1,
                  }
                }
              />
            </div>
            <div
              data-test-button="remove-button"
              onClick={[Function]}
            >
              <i
                className="fas fa-times"
              />
            </div>
          </div>
        </ForwardRef>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Image
              alt="Produto"
              height={80}
              onBrokenImageUrl={[Function]}
              src="http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg"
              width={80}
            />,
            <div>
              <div>
                <span>
                  Smartphone Apple iPhone 7 128GB
                </span>
                <CartSubtotal
                  price={
                    Object {
                      "installmentValue": 389.9,
                      "installments": 10,
                      "value": 3509.1,
                    }
                  }
                />
              </div>
              <div
                data-test-button="remove-button"
                onClick={[Function]}
              >
                <i
                  className="fas fa-times"
                />
              </div>
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "alt": "Produto",
              "height": 80,
              "onBrokenImageUrl": [Function],
              "src": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
              "width": 80,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div>
                  <span>
                    Smartphone Apple iPhone 7 128GB
                  </span>
                  <CartSubtotal
                    price={
                      Object {
                        "installmentValue": 389.9,
                        "installments": 10,
                        "value": 3509.1,
                      }
                    }
                  />
                </div>,
                <div
                  data-test-button="remove-button"
                  onClick={[Function]}
                >
                  <i
                    className="fas fa-times"
                  />
                </div>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span>
                      Smartphone Apple iPhone 7 128GB
                    </span>,
                    <CartSubtotal
                      price={
                        Object {
                          "installmentValue": 389.9,
                          "installments": 10,
                          "value": 3509.1,
                        }
                      }
                    />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Smartphone Apple iPhone 7 128GB",
                    },
                    "ref": null,
                    "rendered": "Smartphone Apple iPhone 7 128GB",
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "price": Object {
                        "installmentValue": 389.9,
                        "installments": 10,
                        "value": 3509.1,
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <i
                    className="fas fa-times"
                  />,
                  "data-test-button": "remove-button",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-times",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": true,
            "rules": Array [
              "
  margin-top: 2px;
  padding: 20px;
  background-color: ",
              "#DAAF00",
              ";
  align-self: stretch;
  display: flex;

  > div: last-child {
    display: flex;
    width: 100%;

    > div:first-child {
      display: flex;
      flex-direction: column;
      justify-content: space-between;
      padding: 0 10px;
      line-height: 0.9em;
      font-weight: 700;
      height: 100%;

      > span {
        color: ",
              "#FFF",
              ";
      }
    }

    > div:last-child {
      margin-left: auto;
      padding: 0 10px;
      cursor: pointer;
    }
  }
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
